Add Racket implementation

tfidfwastaken opened the pull request:
wowsers in Racket this takes ~50 loc including validation and robustness
the interface for making a dfa is also more ez and natural
yes this is a shameless promotion. Use racket people, issa good language thx

tfidfwastaken committed:
Add Racket implementation

wowsers in Racket this takes ~50 loc including validation and
robustness
yes this is a shameless promotion. Use racket people, issa good
language thx

MadhavJivrajani commented:
Haha 
LGTM!

Can you update the README by adding Racket under the list of languages the DFA is implemented in?

tfidfwastaken commented:
Will do. Might make one more change to improve validation.

tfidfwastaken committed:
Improve DFA guarantees and Update Docs

This new update now *guarantees* that no invalid DFA is inserted, ie
to make a DFA your input must have all transitions defined just once
for each state, else the program would not let you create one.

Still remains under 100 loc racket/lisps ftw y u no use ppls

MadhavJivrajani commented:
Looks good, thanks!
